core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java

```diff
@@ -114,6 +115,21 @@ public interface AsyncResponseTransformer<ResponseT, ResultT> {
      */
     void exceptionOccurred(Throwable error);
 
+    /**
+     * todo
+     *
+     * @return
+     */
+    default SplitAsyncResponseTransformer<ResponseT, ResultT> split(long bufferSize) {
+        CompletableFuture<ResultT> future = new CompletableFuture<>();
```

Comment:
Why can't the `split()` method return a `SplittingTransformer` instead of a `SplitAsyncResponseTransformer`? Can't the splitting transformer create the future and make it available to the user in the same way?

Reply:
I was trying to avoid returning the concrete type `SplittingTransformer` and instead return the interface type `SdkPublisher<AsyncResponseTransformer<T>>`. If we return a `SplittingTransformer` here, we would need to make it `@SdkPublicAPI`

Reply:
Is there any reason`SplitAsyncResponseTransformer` doesn't implement `AsyncResponseTransformer` interface?

Reply:
SplitAsyncResponseTransformer is just a holder for the publisher and a completable future, the class implementing AsyncResponseTransformer is SplittingTransformer.IndividualTransformer

Reply:
Should we make it implement AsyncResponseTransformer?

Reply:
no, I don't think so. The usage would be:
```
SplitAsyncResponseTransformer<GetObjectResponse, GetObjectResponse> split = transformer.split(...);
split.publisher().subscribe(downloaderSubscriber);
CompletableFuture<GetObjectResponse> future = split.preparedFuture();
// do whatever we want with the future
GetObjectResponse response = future.join();
```

Reply:
It just seems a bit odd that the static factory method returns a class that doesn't implement the same interface. Can we make it implement it? We did the same for `toPublisher` method

https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/async/AsyncResponseTransformer.java#L235

Reply:
`.split()` is not a static method though, it is an instance method called on a specific instance of an `AsyncResponseTransformer` just like the `split()` method on `AsyncRequestBody`. We could instead return just the SdkPublisher<AsyncResponseTransformer<ResponseT, ResponseT>>` like `AsyncRequestBody.split`. I would need to find another way to deal with the future that needs to be returned from the service call.

Reply:
I see, I'll need to think a bit more on this. Let's keep this conversation open for now.

Reply:
Can we add a TODO to revisit the naming? It's a bit odd that this class is suffixed with AsyncResponseTransformer but doesn't implement it